### PR TITLE
Added unicode_errors arg in loading a vocab file.

### LINKED ISSUE #2569: loading a vocab file doesn't seem to support unicode_errors
#### Problem description

Hi, thank you for providing gensim, which is an amzing lib, for us!

When I tried to read a vocab file generated by native word2vec command, I got an error.
```python
>>> model = gensim.models.KeyedVectors.load_word2vec_format("test_model",unicode_errors='replace',fvocab='test_vocab')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/gensim-3.8.0-py3.7-linux-x86_64.egg/gensim/models/keyedvectors.py", line 1498, in load_word2vec_format
    limit=limit, datatype=datatype)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/gensim-3.8.0-py3.7-linux-x86_64.egg/gensim/models/utils_any2vec.py", line 338, in _load_word2vec_format
    word, count = utils.to_unicode(line).strip().split()
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/gensim-3.8.0-py3.7-linux-x86_64.egg/gensim/utils.py", line 359, in any2unicode
    return unicode(text, encoding, errors=errors)
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 9-10: invalid continuation byte
```

The cause seems that there is no option to work around when some decoding errors happens in the vocab file loading process.

In Python3.x, unicode is replaced with str at the top of utils_any2vec.py.

The default option of str(), built-in function, is
```python
class str(object=b'', encoding='utf-8', errors='strict')
```

So, I propose to make a room to cover some decoding errors for loading a vocab file :)

#### Steps/code/corpus to reproduce

1. Install Anaconda on Ubuntu18.04.
```bash
curl -LO https://repo.anaconda.com/archive/Anaconda3-2019.07-Linux-x86_64.sh
bash Anaconda3-2019.07-Linux-x86_64.sh
```
2 Clone 'gensim'
```bash
git clone https://github.com/RaRe-Technologies/gensim.git
```
3. Install gensim
```bash
cd gensim/
python setup.py install
```
4. Run the following script with test files.
```python
import gensim
model = gensim.models.KeyedVectors.load_word2vec_format("test_model",unicode_errors='replace',fvocab='test_vocab')
```
5. You will get an error like this.
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/gensim-3.8.0-py3.7-linux-x86_64.egg/gensim/models/keyedvectors.py", line 1498, in load_word2vec_format
    limit=limit, datatype=datatype)
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/gensim-3.8.0-py3.7-linux-x86_64.egg/gensim/models/utils_any2vec.py", line 338, in _load_word2vec_format
    word, count = utils.to_unicode(line).strip().split()
  File "/home/ubuntu/anaconda3/lib/python3.7/site-packages/gensim-3.8.0-py3.7-linux-x86_64.egg/gensim/utils.py", line 359, in any2unicode
    return unicode(text, encoding, errors=errors)
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 9-10: invalid continuation byte
```

*Note: These are not generated by word2vec command. these files are for reproducing this error.*

#### Versions

Ubuntu 18.04

```python
>>> import platform; print(platform.platform())
Linux-4.15.0-1044-aws-x86_64-with-debian-buster-sid
>>> import sys; print("Python", sys.version)
Python 3.7.3 (default, Mar 27 2019, 22:11:17)
[GCC 7.3.0]
>>> import numpy; print("NumPy", numpy.__version__)
NumPy 1.17.0
>>> import scipy; print("SciPy", scipy.__version__)
SciPy 1.3.0
>>> import gensim; print("gensim", gensim.__version__)
gensim 3.8.0
>>> from gensim.models import word2vec;print("FAST_VERSION", word2vec.FAST_VERSION)
FAST_VERSION 1

```

[testdata.zip](https://github.com/RaRe-Technologies/gensim/files/3475521/testdata.zip)

### PR DESCRIPTION
**Pull request by wmtzk**
Fix #2569

Simply, added "unicode_errors" option on loading vocab file process.

I've tested and all of them passed with changing numpy version from 1.14.5 to 1.17.0 in tox.ini. 
```
tox -e flake8
tox -e py37-linux
```

The reason I've changed the version of numpy is to avoid the follwoing error, which also happens the HEAD revision of https://github.com/RaRe-Technologies/gensim/commit/f97d0e793faa57877a2bbedc15c287835463eaa9
```bash
tox -e py37-linux
```
```bash
py37-linux recreate: /home/ubuntu/gensim/.tox/py37-linux
py37-linux installdeps: pip>=19.1.1, numpy==1.14.5, scipy==1.1.0, .[test]
py37-linux installed: atomicwrites==1.3.0,attrs==19.1.0,boto==2.49.0,boto3==1.9.202,botocore==1.12.202,certifi==2019.6.16,chardet==3.0.4,Cython==0.29.13,docutils==0.14,gensim==3.8.0,idna==2.8,importlib-metadata==0.19,jmespath==0.9.4,joblib==0.13.2,mock==3.0.5,more-itertools==7.2.0,Morfessor==2.0.2a4,numpy==1.14.5,packaging==19.1,Pillow==6.1.0,pluggy==0.12.0,py==1.8.0,pyparsing==2.4.2,pytest==5.0.1,pytest-rerunfailures==7.0,python-dateutil==2.8.0,python-Levenshtein==0.12.0,pyzmq==18.0.2,requests==2.22.0,s3transfer==0.2.1,scikit-learn==0.21.3,scipy==1.1.0,six==1.12.0,smart-open==1.8.4,testfixtures==6.10.0,torchfile==0.1.0,tornado==6.0.3,urllib3==1.25.3,visdom==0.1.8.8,wcwidth==0.1.7,websocket-client==0.56.0,zipp==0.5.2
py37-linux run-test-pre: PYTHONHASHSEED='1'
py37-linux run-test: commands[0] | python --version
Python 3.7.3
py37-linux run-test: commands[1] | pip --version
pip 19.2.1 from /home/ubuntu/gensim/.tox/py37-linux/lib/python3.7/site-packages/pip (python 3.7)
py37-linux run-test: commands[2] | python -c 'from gensim.models.word2vec import FAST_VERSION; print(FAST_VERSION)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/ubuntu/gensim/gensim/__init__.py", line 5, in <module>
    from gensim import parsing, corpora, matutils, interfaces, models, similarities, summarization, utils  # noqa:F401
  File "/home/ubuntu/gensim/gensim/corpora/__init__.py", line 6, in <module>
    from .indexedcorpus import IndexedCorpus  # noqa:F401 must appear before the other classes
  File "/home/ubuntu/gensim/gensim/corpora/indexedcorpus.py", line 15, in <module>
    from gensim import interfaces, utils
  File "/home/ubuntu/gensim/gensim/interfaces.py", line 21, in <module>
    from gensim import utils, matutils
  File "/home/ubuntu/gensim/gensim/matutils.py", line 1104, in <module>
    from gensim._matutils import logsumexp, mean_absolute_difference, dirichlet_expectation
  File "__init__.pxd", line 918, in init gensim._matutils
ValueError: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 192 from PyObject
ERROR: InvocationError for command /home/ubuntu/gensim/.tox/py37-linux/bin/python -c 'from gensim.models.word2vec import FAST_VERSION; print(FAST_VERSION)' (exited with code 1)
______________________________________________________ summary ______________________________________________________
ERROR:   py37-linux: commands failed

```

#### Versions

Ubuntu 18.04

```python
>>> import platform; print(platform.platform())
Linux-4.15.0-1044-aws-x86_64-with-debian-buster-sid
>>> import sys; print("Python", sys.version)
Python 3.7.3 (default, Mar 27 2019, 22:11:17)
[GCC 7.3.0]
>>> import numpy; print("NumPy", numpy.__version__)
NumPy 1.17.0
>>> import scipy; print("SciPy", scipy.__version__)
SciPy 1.3.0
>>> import gensim; print("gensim", gensim.__version__)
gensim 3.8.0
>>> from gensim.models import word2vec;print("FAST_VERSION", word2vec.FAST_VERSION)
FAST_VERSION 1

```

Thank you for checking that and your consideration!

### CODE REVIEW

**Comment by mpenkov** [comment]
Looks good, can you please add some tests? The tests should fail without your changes, and pass when your changes are active.

**Comment by wmtzk** [comment]
Thank you for checking that. I'll do that. Please give me a moment.

**Comment by wmtzk** [comment]
I've changed the way to take care of fvocab, which is same as model file is doing.
Also I made a few tests for that.
I'd appreciate it if you could check that. Thanks!

File: gensim/models/utils_any2vec.py [hunk 1]
```diff
@@ -335,7 +335,7 @@ def _load_word2vec_format(cls, fname, fvocab=None, binary=False, encoding='utf8'
         counts = {}
         with utils.open(fvocab, 'rb') as fin:
             for line in fin:
-                word, count = utils.to_unicode(line, errors=unicode_errors).strip().split()
+                word, count = utils.to_unicode(line, encoding=encoding, errors=unicode_errors).split(" ")
```
Comment (piskvorky) [inline gensim/models/utils_any2vec.py]: What's up with this sneaky change to `" "`? And dropping `strip`?

This changes the semantics considerably, not just "Added unicode_errors arg".

Comment (wmtzk) [reply gensim/models/utils_any2vec.py]: I had to focus on the change I proposed. I'll fix this soon.

Comment (wmtzk) [reply gensim/models/utils_any2vec.py]: @piskvorky I've reverted except "add unicode_errors arg" and checked the test codes work.

Comment (piskvorky) [reply gensim/models/utils_any2vec.py]: OK. Btw I'm not saying these changes were necessarily bad – I'm saying they were under-documented, with unclear (to me) motivation and implications.

Comment (wmtzk) [reply gensim/models/utils_any2vec.py]: Thanks for your review. The change you pointed out was not well considered. I understood what you said, but I thought I should focus on fixing the reported issue, for now, the other changes should be reported and PRed by another issue and PR. Thanks!

**Comment by mpenkov** [comment]
Your tests fail under Py2.7, can you please take a look?

**Comment by wmtzk** [comment]
Mmm, It's embarrassing to be mentioned such a thing. ^^;  I've checked the tests passed on Python2.7.15 and 3.7.3 env.

**Comment by mpenkov** [comment]
OK, looks good to merge. Congrats on your first contribution, @wmtzk ! :1st_place_medal: 

Good work!

**Comment by wmtzk** [comment]
Thanks!

**Pull request merged**